core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java

```diff
@@ -0,0 +1,220 @@
+ */
+
+package alluxio.master.metastore.rocks;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.master.metastore.BlockStore;
+import alluxio.util.CommonUtils;
+import alluxio.util.io.FileUtils;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.primitives.Longs;
+import org.apache.commons.io.Charsets;
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.ColumnFamilyOptions;
+import org.rocksdb.DBOptions;
+import org.rocksdb.ReadOptions;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.RocksIterator;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Block store backed by RocksDB.
+ */
+public class RocksBlockStore implements BlockStore {
+  private final String mBaseDir;
+
+  private String mDbPath;
+  private RocksDB mDb;
+  private ColumnFamilyHandle mDefaultColumn;
+  private ColumnFamilyHandle mBlockMetaColumn;
+  private ColumnFamilyHandle mBlockLocationsColumn;
+
+  /**
+   * Creates and initializes a rocks block store.
+   */
+  public RocksBlockStore() throws RocksDBException {
+    mBaseDir = Configuration.get(PropertyKey.MASTER_METASTORE_DIR);
+    RocksDB.loadLibrary();
+    initDb(mBaseDir);
+  }
+
+  @Override
+  public Optional<BlockMeta> getBlock(long id) {
+    byte[] length;
+    try {
+      length = mDb.get(mBlockMetaColumn, Longs.toByteArray(id));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+    if (length == null) {
+      return Optional.empty();
+    }
+    return Optional.of(new BlockMeta(id, Longs.fromByteArray(length)));
+  }
+
+  @Override
+  public void putBlock(BlockMeta meta) {
+    try {
+      mDb.put(mBlockMetaColumn, Longs.toByteArray(meta.getId()),
+          Longs.toByteArray(meta.getLength()));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void removeBlock(long id) {
+    try {
+      mDb.delete(mBlockMetaColumn, Longs.toByteArray(id));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void clear() {
+    try {
+      initDb(mBaseDir);
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public List<BlockLocation> getLocations(long id) {
+    RocksIterator iter =
+        mDb.newIterator(mBlockLocationsColumn, new ReadOptions().setPrefixSameAsStart(true));
+    iter.seek(Longs.toByteArray(id));
+    List<BlockLocation> locations = new ArrayList<>();
+    for (; iter.isValid(); iter.next()) {
+      byte[] key = iter.key();
+      long workerId =
+          Longs.fromBytes(key[8], key[9], key[10], key[11], key[12], key[13], key[14], key[15]);
+      String tier = new String(iter.value(), Charsets.UTF_8);
```

Comment:
do we want to store a "string" in the db, or an index to a string? Then, we can avoid the string construction each time. Not sure if it will matter though.